File: app/src/main/java/it/unical/mat/embasp/specializations/clingo/desktop/ClingoDesktopService.java
package it.unical.mat.embasp.specializations.clingo.desktop;

import java.util.List;

import it.unical.mat.embasp.base.Callback;
import it.unical.mat.embasp.base.InputProgram;
import it.unical.mat.embasp.base.OptionDescriptor;
import it.unical.mat.embasp.base.Output;
import it.unical.mat.embasp.platforms.desktop.DesktopService;
import it.unical.mat.embasp.specializations.clingo.ClingoAnswerSets;

/** is an extention of {@link DesktopService} for Clingo 's solver */

public class ClingoDesktopService extends DesktopService {

	public ClingoDesktopService(final String exe_path) {
		super(exe_path);
		load_from_STDIN_option = "";
	}

	@Override
	protected Output getOutput(final String output, final String error) {
		return new ClingoAnswerSets(output, error);
	}

	@Override
	public void startAsync(final Callback callback, final List<InputProgram> programs, final List<OptionDescriptor> options) {
		super.startAsync(callback, programs, options);
	}

	@Override
	public Output startSync(final List<InputProgram> programs, final List<OptionDescriptor> options) {

		return super.startSync(programs, options);
	}
}

File: CSharp version/it/unical/mat/embasp/specializations/clingo/desktop/ClingoDesktopService.cs
namespace it.unical.mat.embasp.specializations.clingo.desktop
{
    using Output = it.unical.mat.embasp.@base.Output;
    using DesktopService = it.unical.mat.embasp.platforms.desktop.DesktopService;

    public class ClingoDesktopService : DesktopService
    {
        public ClingoDesktopService(string exe_path) : base(exe_path) => load_from_STDIN_option = "";

        protected internal override Output GetOutput(string output, string error) => new ClingoAnswerSets(output, error);
    }
}